src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -118,14 +140,42 @@ export const pinTokensWithBalanceAtTop = ({
 };
 
 /**
- * Calculates total USD balance of the provided tokens list.
+ * Calculates total USD balance of the provided UI tokens list.
  *
- * @param tokens - The list of tokens for total USD balance calculation.
- * @returns The sum of tokens USD balance.
+ * @param tokens - The list of UI tokens for total USD balance calculation.
+ * @returns The sum of UI tokens USD balance.
  */
-export const sumTokensUsdBalance = (tokens: TokenUi[]): number =>
+export const sumTokensUiUsdBalance = (tokens: TokenUi[]): number =>
 	tokens.reduce((acc, token) => acc + (token.usdBalance ?? 0), 0);
 
+/**
+ * Calculates total USD balance of mainnet tokens from the provided tokens list.
+ *
+ * @param $tokens - The list of tokens for filtering by network env and total USD balance calculation.
+ * @param $balancesStore - The balances data for the tokens.
+ * @param $exchanges - The exchange rates data for the tokens.
+ * @returns The sum of mainnet tokens USD balance.
+ *
+ */
+export const sumMainnetTokensUsdBalance = ({
+	$tokens,
+	$balances,
+	$exchanges
+}: {
+	$tokens: Token[];
+	$balances: CertifiedStoreData<BalancesData>;
+	$exchanges: ExchangesData;
+}) =>
+	$tokens.reduce((acc, token) => {
+		if (token.network.env === 'mainnet') {
```

Comment:
To avoid looping through `tokens` several times, I added the filtering logic as well as USD calculation to the same function. Let me know if you still think it's worth separating it into several steps.

Reply:
makes sense